tests/MessagePack.SourceGenerator.Tests/MessagePackAnalyzerTests.cs

```diff
@@ -467,4 +468,77 @@ public class Bar
             CodeActionEquivalenceKey = MessagePackCodeFixProvider.AddKeyAttributeEquivanceKey,
         }.RunAsync();
     }
+
+    [Fact]
+    public async Task AddAttributesToMembersOfRecord()
+    {
+        string input = Preamble + /* lang=c#-test */ @"
+[MessagePackObject]
+public record Foo
+{
+    public string {|MsgPack004:Member1|} { get; set; }
+    public string {|MsgPack004:Member2|} { get; set; }
+}
+";
+
+        string output = Preamble + /* lang=c#-test */ @"
+[MessagePackObject]
+public record Foo
+{
+    [Key(0)]
+    public string Member1 { get; set; }
+    [Key(1)]
+    public string Member2 { get; set; }
+}
+";
+        await new VerifyCS.Test
+        {
+            ReferenceAssemblies = ReferenceAssemblies.Net.Net60,
+            CompilerDiagnostics = CompilerDiagnostics.Errors,
+            MarkupOptions = MarkupOptions.UseFirstDescriptor,
+            SolutionTransforms =
+            {
+                static (solution, projectId) =>
+                {
+                    return solution.WithProjectParseOptions(projectId, new CSharpParseOptions(languageVersion: LanguageVersion.CSharp11));
+                },
+            },
+            TestCode = input,
+            FixedCode = output,
+        }.RunAsync();
+    }
+
+    [Fact]
+    public async Task AddAttributesToMembersOfRecordWithPrimaryCtor()
+    {
+        string input = Preamble + /* lang=c#-test */ @"
+[MessagePackObject]
+public record {|MsgPack007:Foo|}(
```

Comment:
@AArnott : Here

Reply:
I think this is a false-positive for records (but a correct one for normal classes, since the semantics of a „class“- primary are different)

Reply:
I like the test as you have it here. Primary constructors for records convert parameters to properties and thus should be serialized. Primary constructors on classes do _not_ convert their parameters to properties and thus should _not_ be serialized. That appears to be what you're asserting here, and all the tests pass, so I'm satisfied. Thank you.